extensions/control-plane/api/management-api/asset-api/src/main/java/org/eclipse/edc/connector/api/management/asset/model/AssetCreationRequestDto.java

```diff
@@ -32,9 +32,21 @@ private AssetCreationRequestDto() {
     }
 
     @JsonIgnore
-    @AssertTrue(message = "no empty property keys")
+    @AssertTrue(message = "no empty property keys and no duplicate keys")
     public boolean isValid() {
-        return properties != null && properties.keySet().stream().noneMatch(it -> it == null || it.isBlank());
+        //check for empty keys
+        boolean validPrivate = privateProperties != null && privateProperties.keySet().stream().noneMatch(it -> it == null || it.isBlank());
+        boolean validPublic = properties != null && properties.keySet().stream().noneMatch(it -> it == null || it.isBlank());
+        return validPrivate && validPublic;
+    }
+
+    @JsonIgnore
+    @AssertTrue(message = "no duplicate keys in properties and private properties")
+    public boolean isMapValid() {
```

Comment:
rename to `containsDistinctKeys` to better reflect what it's doing.

Reply:
renamed it accordingly.

Reply:
had to change the naming again, since the validator only picks up methods with the pattern is***Valid()